src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedNewTests.cs

```diff
@@ -0,0 +1,1128 @@
+        }
+
+        [Fact]
+        public void TestOverloadResolution5()
+        {
+            var comp = CreateCompilation(@"
+class A
+{
+    public A(int i) {}
+}
+class B : A
+{
+    public B(int i) : base(i) {}
+}
+
+class Program
+{
+    static void M(A a, int i) {}
+    static void M(B a, object i) {}
+
+    public static void Main()
+    {
+        M(new(), 1);
+    }
+}
+", options: TestOptions.ReleaseExe).VerifyDiagnostics(
+                // (18,9): error CS0121: The call is ambiguous between the following methods or properties: 'Program.M(A, int)' and 'Program.M(B, object)'
+                //         M(new(), 1);
+                Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("Program.M(A, int)", "Program.M(B, object)").WithLocation(18, 9));
+        }
+
+        [Fact]
+        public void TestAssignmnet()
+        {
+            var comp = CreateCompilation(@"
+class Program
+{
+    public static void Main()
+    {
+        new() = 5;
+    }
+}
+", options: TestOptions.ReleaseExe).VerifyDiagnostics(
+                // (6,9): error CS0131: The left-hand side of an assignment must be a variable, property or indexer
+                //         new() = 5;
+                Diagnostic(ErrorCode.ERR_AssgLvalueExpected, "new()").WithLocation(6, 9));
+        }
+
+        [Fact]
+        public void TestNullableType1()
+        {
+            var comp = CreateCompilation(@"
+using System;
+struct S
+{
+    public S(int i)
+    {
+        Console.Write(i);
+    }
+
+    public static void Main()
+    {
+        S? s = new(43);
+    }
+}
+", options: TestOptions.ReleaseExe);
+
+            CompileAndVerify(comp, expectedOutput: "43");
+        }
+
+        [Fact]
+        public void TestNullableType2()
+        {
+            var comp = CreateCompilation(@"
+using System;
+struct S
+{
+    public static T? M<T>() where T : struct
+    {
+        return new();
+    }
+
+    public static void Main()
+    {
+        Console.Write(M<S>());
+    }
+}
+", options: TestOptions.ReleaseExe);
+
+            CompileAndVerify(comp, expectedOutput: "S");
+        }
+
+        [Fact]
+        public void AsStatement()
+        {
+            var comp = CreateCompilation(@"
+struct S
+{
+    public static void Main()
+    {
+        new(a) { x };
+        new() { x };
+    }
+}
+", options: TestOptions.ReleaseExe).VerifyDiagnostics(
+                // (6,13): error CS0103: The name 'a' does not exist in the current context
+                //         new(a) { x };
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "a").WithArguments("a").WithLocation(6, 13),
+                // (7,9): error CS0201: Only assignment, call, increment, decrement, await, and new object expressions can be used as a statement
+                //         new() { x };
+                Diagnostic(ErrorCode.ERR_IllegalStatement, "new() { x }").WithLocation(7, 9)
+                );
+        }
+    }
+}
```

Comment:
Can you please add tests to show how the semantic model works? In normal (non-error) cases, e.g. for the whole expression and its argument expressions, but also inside the object initializer, and inside the object initializer when there is no target type (error scenario), for example
``` c#
    M(new() { X = <bind>N()</bind> }); // M not found, but N is found. Also try when M is found but no X is found
```
How does `GetTypeInfo` work on the subexpression `N()`? Even if it doesn't work, it would be nice to have a test documenting the current (expected?) behavior.

Reply:
If it's ok I'll address these in the follow-up PR.